Glaze doesn't build with VS2022

If I create default console project with `#include <glaze/glaze.hpp>` and try to compile it fails with the error:

```
1>test-glaze.cpp
1>D:\work-pps\glaze\include\glaze\util\parse.hpp(1310,39): error C4146: unary minus operator applied to unsigned type, result still unsigned
```
this is the code that causes the error:

```
               // Negate in unsigned arithmetic to avoid signed overflow when acc == limit
               // (e.g., when parsing -2147483648, acc = 2147483648u for int32)
               value = static_cast<I>(-acc);
```

Note, I do not have warnings as errors enabled. It's default console projects with no options changes/updated. It plainly errors out. IMO, this cast should be properly handled to avoid warnings there.

--- a/include/glaze/util/parse.hpp
+++ b/include/glaze/util/parse.hpp
@@ -1298,16 +1298,23 @@ namespace glz
          if constexpr (is_signed) {
             // The largest magnitude we can represent in a negative value is (max + 1)
             // since -(min()) = max() + 1.
-            U limit = static_cast<U>((std::numeric_limits<I>::max)()) + 1U;
+            const U limit = static_cast<U>((std::numeric_limits<I>::max)()) + 1U;
             if (negative) {
                if (acc > limit) {
                   result.ec = std::errc::result_out_of_range;
                   result.ptr = first;
                   return result;
                }
+#if defined(_MSC_VER) && !defined(__clang__)
+#pragma warning(push)
+#pragma warning(disable : 4146)
+#endif
                // Negate in unsigned arithmetic to avoid signed overflow when acc == limit
                // (e.g., when parsing -2147483648, acc = 2147483648u for int32)
                value = static_cast<I>(-acc);
+#if defined(_MSC_VER) && !defined(__clang__)
+#pragma warning(pop)
+#endif
             }
             else {
                if (acc > static_cast<U>((std::numeric_limits<I>::max)())) {

